File: android/src/androidTest/java/org/ligi/gobandroidhd/unittest/TheMarkerUtilLetterFinder.java
package org.ligi.gobandroidhd.unittest;

import android.test.suitebuilder.annotation.SmallTest;

import org.ligi.gobandroid_hd.logic.markers.util.MarkerUtil;
import org.ligi.gobandroidhd.base.MarkerTestBase;

import static org.assertj.core.api.Assertions.assertThat;

public class TheMarkerUtilLetterFinder extends MarkerTestBase {

    @SmallTest
    public void testFindFirst() {
        final String firstFreeLetter = MarkerUtil.findNextLetter(markerList("C", "B"));
        assertThat(firstFreeLetter).isEqualTo("A");
    }


    @SmallTest
    public void testFindLast() {
        final String firstFreeLetter = MarkerUtil.findNextLetter(markerList("A", "B"));
        assertThat(firstFreeLetter).isEqualTo("C");
    }


    @SmallTest
    public void testFindGap() {
        final String firstFreeLetter = MarkerUtil.findNextLetter(markerList("A", "C"));
        assertThat(firstFreeLetter).isEqualTo("B");
    }


    @SmallTest
    public void testSurviveNumbers() {
        final String firstFreeLetter = MarkerUtil.findNextLetter(markerList("A", "1"));
        assertThat(firstFreeLetter).isEqualTo("B");
    }


    @SmallTest
    public void testFindZ() {
        final String firstFreeLetter = MarkerUtil.findNextLetter(markerList("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y"));
        assertThat(firstFreeLetter).isEqualTo("Z");
    }
}

File: android/src/test/java/org/ligi/gobandroid_hd/TheMarkerUtilLetterFinder.kt
package org.ligi.gobandroid_hd

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.ligi.gobandroid_hd.logic.markers.functions.findNextLetter

class TheMarkerUtilLetterFinder {

    @Test
    fun testFindFirst() {
        val firstFreeLetter = markerList("C", "B").findNextLetter()
        assertThat(firstFreeLetter).isEqualTo("A")
    }

    @Test
    fun testFindLast() {
        val firstFreeLetter = markerList("A", "B").findNextLetter()
        assertThat(firstFreeLetter).isEqualTo("C")
    }

    @Test
    fun testFindGap() {
        val firstFreeLetter = markerList("A", "C").findNextLetter()
        assertThat(firstFreeLetter).isEqualTo("B")
    }


    @Test
    fun testSurviveNumbers() {
        val firstFreeLetter = markerList("A", "1").findNextLetter()
        assertThat(firstFreeLetter).isEqualTo("B")
    }


    @Test
    fun testFindZ() {
        val firstFreeLetter = markerList(('A'..'Y').toList().map(Char::toString)).findNextLetter()
        assertThat(firstFreeLetter).isEqualTo("Z")
    }


    @Test
    fun testFindAfterZ() {
        val firstFreeLetter = markerList(('A'..'Z').toList().map(Char::toString)).findNextLetter()
        assertThat(firstFreeLetter).isEqualTo("a")
    }
}